tests/Microsoft.Identity.Test.Integration.netcore/HeadlessTests/Agentic.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System;
+using System.Diagnostics;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Extensibility;
+using Microsoft.Identity.Test.LabInfrastructure;
+using Microsoft.Identity.Test.Unit;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Integration.HeadlessTests
+{
+    [TestClass]
+    public class Agentic
+    {
+        const string ClientId = "d15884b6-a447-4dd5-a5a5-a668c49f6300"; // agent app
+        const string TenantId = "31a58c3b-ae9c-4448-9e8f-e9e143e800df";
+        const string AgentIdentity = "d84da24a-2ea2-42b8-b5ab-8637ec208024";
+        const string UserUpn = "aui1@msidlabtoint.onmicrosoft.com";
+        private const string TokenExchangeUrl = "api://AzureADTokenExchange/.default";
+
+        [TestMethod]
+        public async Task AgentUserIdentityGetsTokenForGraphTest()
+        {
+            await AgentUserIdentityGetsTokenForGraphAsync().ConfigureAwait(false);
```

Comment:
Why are these methods called twice? Maybe assert if the result was from IDP or cache if that is the reason.

Reply:
Yes, good point, updated.

Reply:
I added an AcquireTokenSilent option to the user fic call.